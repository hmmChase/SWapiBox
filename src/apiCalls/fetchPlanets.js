import { doFetch } from './doFetch';

export const fetchPlanets = async () => {
  try {
    const root = 'https://swapi.co/api';
    const planetsData = await doFetch(`${root}/planets`);
    const parsedPlanetsData = planetsData.results.map(async planet => {
      const residentsData = planet.residents.map(async residentURL => {
        return await fetchResidents(residentURL);
      });
      const resolveResidents = await Promise.all(residentsData);
      return {
        name: planet.name,
        terrain: planet.terrain,
        population: planet.population,
        climate: planet.climate,
        residents: resolveResidents.length
          ? resolveResidents.join(', ')
          : 'unknown'
      };
    });
    return await Promise.all(parsedPlanetsData);
  } catch (error) {
    throw new Error(error.message);
  }
};

export const fetchResidents = async residentURL => {
  try {
    const residentData = await doFetch(residentURL);
    return residentData.name;
  } catch (error) {
    throw new Error(error.message);
  }
};
