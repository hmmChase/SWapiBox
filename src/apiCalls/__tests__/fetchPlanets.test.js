import { fetchPlanets } from "../fetchPlanets";

describe('planetData', () => {

  it('fetchPlanets is defined', () => {
    expect(fetchPlanets).toBeDefined();
  });

});