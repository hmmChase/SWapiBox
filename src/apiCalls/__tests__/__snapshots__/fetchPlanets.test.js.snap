// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`planetData fetchPlanets matches snapshot 1`] = `[Function]`;
